Add copy button when creating new application password Fix/62019

circlecube opened the pull request:
<!--
Hi there! Thanks for contributing to WordPress!

Pull Requests in this GitHub repository **must** be linked to a ticket in the WordPress Core Trac instance (https://core.trac.wordpress.org), and are only used for code review. **No pull requests will be merged on GitHub.**

See the WordPress Handbook page on using PRs for Code Review more information: https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/

If this is your first time contributing, you may also find reviewing these guides first to be helpful:
- FAQs for New Contributors: https://make.wordpress.org/core/handbook/tutorials/faq-for-new-contributors/
- Contributing with Code Guide: https://make.wordpress.org/core/handbook/contribute/
- WordPress Coding Standards: https://make.wordpress.org/core/handbook/best-practices/coding-standards/
- Inline Documentation Standards: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/
- Browser Support Policies: https://make.wordpress.org/core/handbook/best-practices/browser-support/
- Proper spelling and grammar related best practices: https://make.wordpress.org/core/handbook/best-practices/spelling/
-->

<!-- Insert a description of your changes here -->

Trac ticket: https://core.trac.wordpress.org/ticket/62019

Updating the existing patch to capitalize the button text 'copy > 'Copy' and making the text translatable.

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

circlecube committed:
apply 62019 patch from dhruvang21

circlecube committed:
update copy button capitalization and make translatable

circlecube committed:
add ClipboardJS to list of global vars for linting

circlecube committed:
update application password e2e test to reflect changes

jeremyfelt commented on src/js/_enqueues/admin/user-profile.js:
```diff
@@ -346,6 +348,27 @@
+		}, 3000 );
+
+		// Handle success audible feedback.
+		wp.a11y.speak( __( 'Site information has been copied to your clipboard.' ) );
```
```suggestion
		wp.a11y.speak( __( 'Application password has been copied to your clipboard.' ) );
```

Let's change this to reference the application password fields.

circlecube committed:
Update src/js/_enqueues/admin/user-profile.js

Co-authored-by: Jeremy Felt <jeremy.felt@gmail.com>

circlecube committed:
Merge branch 'trunk' into fix/62019

* trunk:
  Build Tools: Allow easier customization of the .env file.
  Coding Standards: Avoid using confusing `!` condition in Media Library selection check.
  REST API: Only check password value in query parameters while checking post permissions.
  Media: Add Ctrl/Command + Enter shortcut to insert selected Media Library items.
  REST API: Support exact search in the REST API posts endpoint.
  Script Loader: Remove unused array_merge.
  REST API: Automatically populate targetHints for the Allow header.

jeremyfelt commented:
Things are looking good so far, @circlecube, thanks!

A couple more notes:

1. We should apply these changes to `wp-admin/profile.php` as well. I think folks are more likely to be adding application passwords through that screen rather than while editing another user.
2. We can probably add a bit of spacing above "Copy" on mobile. (See screenshot)

![CleanShot 2024-09-17 at 15 42 28@2x](https://github.com/user-attachments/assets/88b69339-4400-429f-a72b-7842e37b5cd0)

circlecube commented:
> 1. We should apply these changes to wp-admin/profile.php as well. I think folks are more likely to be adding application passwords through that screen rather than while editing another user.

The `wp-admin/profile.php` file loads up `wp-admin/user-edit/php` i.e. these changes are on the profile page now.

> 2. We can probably add a bit of spacing above "Copy" on mobile.

Good catch @jeremyfelt, looking at this now.

jeremyfelt commented:
> The wp-admin/profile.php file loads up wp-admin/user-edit/php i.e. these changes are on the profile page now.

Ha! I **swear** this was not working, but there it is. My fault!

circlecube committed:
adjust spacing between input and copy button

(this only applies in some small screen sizes when this wraps into multiple lines)

circlecube commented:
Screenshots of the mobile layout with a tiny CSS tweak:
![application-secret-copy-mobile-small](https://github.com/user-attachments/assets/f5fe538a-c190-449a-84bf-a1c8dc2b1369)
![application-secret-copy-mobile](https://github.com/user-attachments/assets/2ecb6373-50a3-4791-aad4-0ab26a995a51)
